src/BusinessLayer/InternetReporting.cpp

```diff
@@ -27,31 +27,95 @@ InternetReporting::InternetReporting(I_CommunicationService& commService,
 
     : communicationService_(commService)
     , view_(view)
-    , packetNum(0)
+    , signalMapper_(this)
+    , packetNum_(0)
+    , readTimer_(new QTimer())
+    , forwardPeriod_(500)
+    , sendContinuously_(false)
 {
     dataContainerList.push_back(&dataContainer0);
     dataContainerList.push_back(&dataContainer1);
-    connect(&view_, SIGNAL(sendAll(int)), this, SLOT(sendAll(int)));
+    connect(&view_, SIGNAL(toggleSendContinuously()), this, SLOT(toggleSendContinuously()));
+    connect(&view_, SIGNAL(setPacketNum(int)), this, SLOT(setPacketNum(int)));
+    connect(readTimer_.data(), SIGNAL(timeout()), this, SLOT(sendAll()));
+    readTimer_->setInterval(forwardPeriod_);
+}
+
+void InternetReporting::toggleSendContinuously()
+{
+    sendContinuously_ = !sendContinuously_;
+
+    if (sendContinuously_)
+    {
+        readTimer_->start();
+    }
+    else
+    {
+        readTimer_->stop();
+    }
+
+    view_.setSendContinuouslyText(sendContinuously_);
```

Comment:
The logic of changing the send continously text should stay contained within the view. Try hooking up the getSendContinuouslyButton() signal in the view to setSendContinouslyText() within `InternetView.cpp`